src/libawkward/array/NumpyArray.cpp

```diff
@@ -1574,7 +1577,7 @@ namespace awkward {
             dtype_ == util::dtype::complex64  ||
             dtype_ == util::dtype::complex128  ||
             dtype_ == util::dtype::complex256  ||
-            dtype_ == util::dtype::boolean  ||
+            rawother->dtype() == util::dtype::boolean  ||
```

Comment:
reported by static analyser

Reply:
It was a good catch! This had been wrong and is now fixed.